### PR TITLE
Fix crash in median over Nullable(Decimal128)

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Crash in median over Nullable(Decimal128)

### CODE REVIEW

**Comment by 4ertus2** [comment]
There's no alignment in AllocatorWithStackMemory. It leads to crash in attepmt to push_back __int128 into PODArrayWithStackMemory

**Comment by alexey-milovidov** [comment]
Also we can replace `__int128` to more portable `boost::cpp_int`. It may lead to performance degradation, but most likely not.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Asked a question in Telegram chat.

File: dbms/src/Common/PODArray.h [hunk 1]
```diff
@@ -636,6 +636,6 @@ using PaddedPODArray = PODArray<T, initial_bytes, TAllocator, 15, 16>;
 template <typename T, size_t inline_bytes,
           size_t rounded_bytes = integerRoundUp(inline_bytes, sizeof(T))>
 using PODArrayWithStackMemory = PODArray<T, rounded_bytes,
-    AllocatorWithStackMemory<Allocator<false>, rounded_bytes>>;
+    AllocatorWithStackMemory<Allocator<false>, rounded_bytes, ((sizeof(T) < 16) ? sizeof(T) : 16)>>;
```
Comment (alexey-milovidov) [inline dbms/src/Common/PODArray.h]: There is also `alignof` keyword.

Comment (alexey-milovidov) [reply dbms/src/Common/PODArray.h]: I'll change.

**Pull request merged**